### DISCUSSION TITLE
Do We Still Need terminationDrainDuration and holdApplicationUntilProxyStarts After Enabling Kubernetes Native Sidecars in Istio?

### DISCUSSION THREAD

**Discussion by darmris** [istio/istio#53174] [General]
I've recently enabled **Kubernetes native sidecars** in my cluster by setting the `ENABLE_NATIVE_SIDECARS` environment variable, and I understand that Kubernetes now manages the lifecycle of the sidecar containers (like Istio's Envoy proxy). This includes controlling the startup and shutdown order of the application and sidecar containers.

Given that Kubernetes ensures:

- The sidecar proxy starts before the main application.

- The sidecar remains running until the application container has completely shut down during termination.

I’m wondering if it’s still necessary to use Istio-specific configurations like:

1. `terminationDrainDuration`: This parameter ensures that the Envoy proxy continues to drain in-flight connections during shutdown before terminating.

2. `holdApplicationUntilProxyStarts`: This ensures that the application container only starts after the sidecar proxy is fully ready to handle traffic.

My Questions:
1. **Is `terminationDrainDuration` still necessary** when Kubernetes is already managing the sidecar shutdown order with native sidecars, or can we rely on Kubernetes to handle graceful shutdowns without this setting?

2. **Does `holdApplicationUntilProxyStarts` provide any additional value** when ENABLE_NATIVE_SIDECARS is enabled, or does Kubernetes fully ensure that the sidecar proxy is ready before traffic hits the application?

Context:

- I’m using Istio in a production environment with both stateless and stateful services.

- My goal is to simplify the configuration if possible, but I still want to ensure that the proxy is fully handling traffic before the application starts and is gracefully terminating connections during shutdown.

Thanks in advance for any insights!

**Comment by howardjohn** [comment]
Neither is needed, and I think both are entirely ignored when this is set.

Both of those variables were effectively workarounds for the lack of native sidecars in Istio
